### PR TITLE
[Merged by Bors] - chore: make a global instance local in ModelTheory.Order

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/ModelTheory/Order.lean [hunk 1]
```diff
@@ -358,10 +358,14 @@ instance : @OrderedStructure L M _ (L.leOfStructure M) _ := by
...
-    DecidableRel (@LE.le M (L.leOfStructure M)) := by
-  letI := L.leOfStructure M
-  exact h
+/-- The order structure on an ordered language is decidable. -/
+-- This should not be a global instance,
+-- because it will match with any `LE` typeclass search
+@[local instance]
+def decidableLEOfStructure
+    [h : DecidableRel (fun (a b : M) => Structure.RelMap (leSymb : L.Relations 2) ![a,b])] :
+    have := L.leOfStructure M
```
Comment (sgouezel) [inline Mathlib/ModelTheory/Order.lean:367]: I'd prefer a `let` or a `letI` since you're creating data here.

Comment (kim-em) [reply Mathlib/ModelTheory/Order.lean:367]: (Please remember to set PRs at `awaiting-author`, @sgouezel, otherwise they may go stale. :-)

Comment (kim-em) [reply Mathlib/ModelTheory/Order.lean]: Done.

**Comment by sgouezel** [comment]
bors r+
Thanks

**Pull request closed without merging**